src/adcm_pytest_plugin/fixtures.py

```diff
@@ -294,37 +297,64 @@ def _remove_hosts(adcm_cli: ADCMClient):
             job.wait(timeout=60)
 
 
+def _adcm_is_upgradable(request) -> bool:
+    if hasattr(request, "param") and request.param:
+        return True
+    return False
+
+
 ##################################################
 #                  S D K
 ##################################################
 @allure.title("[MS] ADCM Container")
 @pytest.fixture(scope="module")
-def adcm_ms(image, cmd_opts, request, adcm_api_credentials) -> Generator[ADCM, None, None]:
+def adcm_ms(image, cmd_opts, request, adcm_api_credentials, adcm_is_upgradable_ms: bool) -> Generator[ADCM, None, None]:
     """Runs adcm container from the previously initialized image.
     Operates '--dontstop' option.
     Returns authorized instance of ADCM object
     """
-    yield from _adcm(image, cmd_opts, request, adcm_api_credentials)
+    yield from _adcm(image, cmd_opts, request, adcm_api_credentials, upgradable=adcm_is_upgradable_ms)
 
 
 @allure.title("[FS] ADCM Container")
 @pytest.fixture(scope="function")
-def adcm_fs(image, cmd_opts, request, adcm_api_credentials) -> Generator[ADCM, None, None]:
+def adcm_fs(image, cmd_opts, request, adcm_api_credentials, adcm_is_upgradable_fs: bool) -> Generator[ADCM, None, None]:
     """Runs adcm container from the previously initialized image.
     Operates '--dontstop' option.
     Returns authorized instance of ADCM object
     """
-    yield from _adcm(image, cmd_opts, request, adcm_api_credentials)
+    yield from _adcm(image, cmd_opts, request, adcm_api_credentials, upgradable=adcm_is_upgradable_fs)
 
 
 @allure.title("[SS] ADCM Container")
 @pytest.fixture(scope="session")
-def adcm_ss(image, cmd_opts, request, adcm_api_credentials) -> Generator[ADCM, None, None]:
+def adcm_ss(image, cmd_opts, request, adcm_api_credentials, adcm_is_upgradable_ss: bool) -> Generator[ADCM, None, None]:
     """Runs adcm container from the previously initialized image.
     Operates '--dontstop' option.
     Returns authorized instance of ADCM object
     """
-    yield from _adcm(image, cmd_opts, request, adcm_api_credentials)
+    yield from _adcm(image, cmd_opts, request, adcm_api_credentials, upgradable=adcm_is_upgradable_ss)
+
+
+@allure.title("[SS] ADCM upgradable flag")
+@pytest.fixture(scope="session")
+def adcm_is_upgradable_ss(request) -> bool:
+    """Set flag that controls either ADCM will be upgradable or not"""
+    return _adcm_is_upgradable(request)
+
+
+@allure.title("[MS] ADCM upgradable flag")
+@pytest.fixture(scope="module")
+def adcm_is_upgradable_ms(request) -> bool:
+    """Set flag that controls either ADCM will be upgradable or not"""
+    return _adcm_is_upgradable(request)
+
+
+@allure.title("[FS] ADCM upgradable flag")
+@pytest.fixture()
+def adcm_is_upgradable_fs(request) -> bool:
+    """Set flag that controls either ADCM will be upgradable or not"""
+    return _adcm_is_upgradable(request)
```

Comment:
Why not use a single session-scoped fixture?

Reply:
In that case we'll alter all levels of adcm fixture, so each adcm_fs will be upgradable

Reply:
Nope.
Only adcm_xx fixtures of the given module or package.

Reply:
Got it, fixed